a3p-integration/proposals/b:restart-vats/package.json

```
@@ -0,0 +1,17 @@
+{
+    "agoricProposal": {
+        "type": "/agoric.swingset.CoreEvalProposal",
+        "source": "build",
+        "buildScript": "../packages/builders/scripts/vats/restart-vats.js"
```

hm. cross-package reference. Did you consider supporting this sort of thing?

```suggestion
        "buildScript": "@agoric/builders/scripts/vats/restart-vats.js"
```

---

I'm actually confused as to which context this import is resolved in, given the `../packages`. I know the upgrade core proposal mechanism implemented by `cosmic-swingset` supports the `@agoric/builders` form, and it would be confusing if this type of proposal didn't.

More generally however, I'm wondering if a "resolve in the context of agoric-sdk" approach is correct for core evals. And where does the agoric-sdk even come from, the one inside the image? Consider 3rd party contracts that only depend on agoric-sdk packages through NPM installed packages, no checkout of `agoric-sdk`. How should their version of `a3p-integration` in their own repo resolve the build script?

---

Right now this is just relative to where the `synthetic-chain` is run, so `a3p-integration`. I could do some work to make it resolves against NPM packages, but can we call that future work?

>How should their version of a3p-integration in their own repo resolve the build script?

Right now it's the relative path. If we went the package resolution route, it would be the package export. But I don't think we should require them to make their build script an export from a package. It's just a file.

---

>normally you want to generate a core eval from your local repo, which may not be agoric-sdk.

Exactly. That's why I think it best to leave this to filesystem path resolution. Another repo might have `a3p-integration` as its home for this tooling, or not. It'll just be somewhere in the repo. And the build script will be referenced relative to that.

Oh, maybe here's the confusion: this buildScript is executed before any Docker comes into play. It's a convention of the `synthetic-chain` tooling that it builds the script and puts the output into the proposal directory, so that it's there during the image building.

---

> Oh, maybe here's the confusion: this buildScript is executed before any Docker comes into play. It's a convention of the `synthetic-chain` tooling that it builds the script and puts the output into the proposal directory, so that it's there during the image building.

Ah that's where I was confused, and yes this is the behavior I expect. In this case I am fine with relative path, they actually make more sense.

---

> I could do some work to make it resolves against NPM packages, but can we call that future work?

Yes.

And I'm content without an issue until motivation is more pressing.

---

Now running into,
```
agoric: run: running /home/runner/work/agoric-sdk/agoric-sdk/packages/builders/scripts/vats/restart-vats.js
agoric: (Error#1)
Error#1: Cannot find package '@agoric/deploy-script-support' imported from /home/runner/work/agoric-sdk/agoric-sdk/packages/builders/scripts/vats/restart-vats.js
```

So it's not "just a file". I could add `@agoric/deploy-script-support` to the a3p-integration package.json, but it'll surprise someone again when they depend on something else.

I think a3p-integration needs to be part of the repo workspace, so when it runs these scripts from the repo they can use anything in the repo.

---

I've got a commit to do that, but it runs into a new problem with https://github.com/Agoric/agoric-3-proposals/blob/main/packages/synthetic-chain/cli.ts#L53-L56

I'll mark this as Draft until I get that working in a 0.0.4-3 or later